crypto/ec/ecx_meth.c

```diff
@@ -334,7 +334,6 @@ static int ecd_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
 static int ecx_set_priv_key(EVP_PKEY *pkey, const unsigned char *priv,
                             size_t len)
 {
-    /* TODO(3.0): We should pass a libctx here */
```

Comment:
This one is potentially problematic and should probably not be part of this PR.

Reply:
I've added a code to get the libctx from the pkey's keymgmt if it is set. Not sure it will be so if this function is called though. However that is what we can do here.

Reply:
After tracking down where this is called from maybe this is not a good idea..

It seems to be called only from legacy paths for EVP_PKEY_new_raw_private_key_ex().
Provider keys would just use fromdata. The legacy path that uses this does
pkey = EVP_PKEY_new();
pkey_set_type(pkey, e,nidtype, strtype, -1, NULL) //where NULL is the keymanager.
pkey->ameth->set_priv_key().

So even though EVP_PKEY_new_raw_private_key_ex() passes a libctx, propq there is no passing of this down..
unless we had a variant of set_priv_key that passed the libctx,propq then it wont work..
@mattcaswell should we care about this?

Reply:
looks like the set_priv_key was added in 1_1_1 so we cant just change it directly.

Reply:
I'm not sure why any of the code in this file needs a libctx. It's all legacy ameth/pmeth stuff which is typically not libctx aware. My guess is this was introduced at some intermediate stage where we didn't have all the encoder/decoder stuff etc in place yet???

Reply:
I do not think the current state as being added in the PR is harmful, it is just most probably a dead code.